packages/components/src/internal/components/domainproperties/dataset/DatasetPropertiesAdvancedSettings.test.tsx

```diff
@@ -3,11 +3,27 @@ import { render } from '@testing-library/react';
 
 import { waitFor } from '@testing-library/dom';
 
+import { List } from 'immutable';
+
 import { NEW_DATASET_MODEL_WITHOUT_DATASPACE } from '../../../../test/data/constants';
 
+import { QueryInfo } from '../../../../public/QueryInfo';
+
 import { AdvancedSettings, DatasetSettingsInput, DatasetSettingsSelect } from './DatasetPropertiesAdvancedSettings';
 import { DatasetModel } from './models';
 
+jest.mock('../../../query/api', () => ({
+    ...jest.requireActual('../../../query/api'),
+    selectRowsDeprecated: () =>
+        Promise.resolve({
+            key: 'test',
+            models: { test: {} },
+            orderedModels: { test: List() },
+            queries: { test: QueryInfo.fromJsonForTests({}) },
+            rowCount: 0,
+        }),
```

Comment:
This block of code is repeated in a few tests, maybe we can make a utility to reduce some of the boiler plate so it will look something like:

```suggestion
    selectRowsDeprecated: createMockSelectRowsResponse(),
```

Then in the utility for that method you can have a small doc string the explains how to use it:

```typescript
/**
* Use this method with jest.mock, and jest.requireActual in order to prevent network requests from
* occurring in your tests. See DatasetPropertiesAdvancedSettings.test.tsx for an example.
*/
export function createMockSelectRowsResponse() {}
```

Reply:
Done